codemods/run-codemod.ts

```diff
@@ -0,0 +1,191 @@
+        }
+
+        console.log(
+            `Running codemod: ${codemod} on output path: ${outputPath}`
+        );
+
+        try {
+            execSync(command, { stdio: "inherit" });
+            console.log(
+                `Codemod ${codemod} executed successfully on ${outputPath}.`
+            );
+        } catch (error) {
+            console.error(
+                `Error executing codemod ${codemod}: ${error.message}`
+            );
+        }
+    });
+}
+
+// Choose path for codemod use
+async function chooseOutputPath(): Promise<string | null> {
+    const options = [
+        { name: "Src", value: predefinedPaths.Src },
+        { name: "Codebase", value: predefinedPaths.Codebase },
+        { name: "TestData", value: predefinedPaths.TestData },
+        { name: "Custom", value: "custom" },
+        { name: "Exit", value: "exit" },
+    ];
+
+    const selectedOption = await select({
+        message: "Choose an output path:",
+        choices: options,
+    });
+
+    if (selectedOption === "exit") {
+        console.log("Exiting...");
+        return null;
+    }
+
+    if (selectedOption === "custom") {
+        const customPath = await input({ message: "Enter your custom path:" });
+        const resolvedPath = path.resolve(customPath);
+
+        if (!fs.existsSync(resolvedPath)) {
+            console.error(
+                `The path "${resolvedPath}" does not exist. Please try again.`
+            );
+            return null;
+        }
+
+        return resolvedPath;
+    }
+
+    return selectedOption;
+}
+
+async function chooseTheme(): Promise<string | null> {
+    const themeOptions = [
+        { name: "LifeSG", value: "lifesg" },
+        { name: "BookingSG", value: "bookingsg" },
+    ];
+
+    const selectedTheme = await select({
+        message: "Select the theme to follow for Color migration:",
+        choices: themeOptions,
+    });
+
+    return selectedTheme;
+}
+
+// Choose and run the codemods
+async function main(): Promise<void> {
+    const codemods = listCodemods();
+    if (codemods.length === 0) {
+        console.log("No codemod scripts found.");
+        return;
+    }
+
+    const selectedCodemods = await checkbox({
+        message: "Select codemods to run (use space to select multiple):",
+        choices: codemods,
+    });
+
+    if (selectedCodemods.includes("exit")) {
+        console.log("Exiting...");
+        return;
+    }
+
+    const codemodConfirmed = await confirm({
+        message: `You selected "${selectedCodemods.join(
+            ", "
+        )}". Do you want to proceed?`,
+        default: true,
+    });
+
+    if (!codemodConfirmed) {
+        console.log("Codemod selection canceled. Exiting...");
+        return;
+    }
+
+    let selectedTheme: string | null = null;
+
+    if (selectedCodemods.includes("migrate-colour")) {
+        selectedTheme = await chooseTheme();
+
+        const themeConfirmed = await confirm({
```

Comment:
I think having the confirmation in between can get excessive, the user can do crtl+c if they wish to abort

suggest to only prompt for confirmation at the end before triggering `runCodemods()`